### PR TITLE
reduce capi rbac

### PR DESCRIPTION
**Pull request by KodieGlosserIBM**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go [hunk 1]
```diff
@@ -2687,17 +2687,15 @@ func reconcileCAPIManagerRole(role *rbacv1.Role) error {
...
 		{
 			APIGroups: []string{""},
 			Resources: []string{
 				"configmaps",
-				"events",
-				"nodes",
 				"secrets",
 			},
-			Verbs: []string{"*"},
+			Verbs: []string{"list", "get", "watch"},
```
Comment (csrwng) [inline hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go:2698]: picky nit: can you keep these in the same order for consistency?

Comment (KodieGlosserIBM) [reply hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go]: yes, added!

**Comment by csrwng** [comment]
/hold
taking another look at this one

**Comment by csrwng** [comment]
@KodieGlosserIBM @a-dsouza found the issue. The CAPI manager fails to delete the hostedcontrolplane when deprovisioning a cluster:

```
E0216 00:47:10.185291       1 controller.go:317] controller/cluster "msg"="Reconciler error" "error"="failed to delete hypershift.openshift.io/v1beta1, Kind=HostedControlPlane \"cewong-guest\" for Cluster \"cewong-guest-s59z2\" in namespace \"clusters-cewong-guest\": hostedcontrolplanes.hypershift.openshift.io \"cewong-guest\" is forbidden: User \"system:serviceaccount:clusters-cewong-guest:cluster-api\" cannot delete resource \"hostedcontrolplanes\" in API group \"hypershift.openshift.io\" in the namespace \"clusters-cewong-guest\"" "name"="cewong-guest-s59z2" "namespace"="clusters-cewong-guest" "reconciler group"="cluster.x-k8s.io" "reconciler kind"="Cluster"
```

**Comment by a-dsouza** [comment]
@csrwng Thanks for the assistance--opened a new PR for this (#2173), so feel free to close this one!

**Pull request closed without merging**